powerauth-client-model/src/main/java/com/wultra/security/powerauth/client/model/request/OperationTemplateCreateRequest.java

```diff
@@ -37,42 +42,58 @@ public class OperationTemplateCreateRequest {
     /**
      * The name of the operation template.
      */
+    @Schema(description = "The name of the operation template")
+    @NotBlank(message = "Operation template name must not be empty when creating operation template")
     private String templateName;
 
     /**
      * The type of the operation that is created based on the template.
      */
+    @Schema(description = "The type of the operation that is created based on the template")
+    @NotBlank(message = "Operation type must not be empty when creating operation template")
     private String operationType;
 
     /**
      * Template for the operation data.
      */
+    @Schema(description = "Template for the operation data")
+    @NotBlank(message = "Operation template data must not be empty when creating operation template")
     private String dataTemplate;
 
     /**
      * Allowed signature types.
      */
     @JsonSetter(nulls = Nulls.SKIP)
-    private final List<SignatureType> signatureType = new ArrayList<>();
+    @Schema(description = "Allowed signature types")
+    @NotEmpty(message = "Template signature types must contain at least one value")
+    private final List<@NotNull SignatureType> signatureType = new ArrayList<>();
 
     /**
-     * How many failed attempts should be allowed for th operation.
+     * How many failed attempts should be allowed for the operation.
      */
+    @Schema(description = "How many failed attempts should be allowed for the operation")
+    @NotNull(message = "Template expiration value must not be null when creating operation template")
```

Comment:
Wrong parameter reference
```suggestion
    @NotNull(message = "Template maximum allowed failure count must not be null when creating operation template")
```